File: solutions/ruby/space-age/1/space_age.rb
=begin
Write your code for the 'Space Age' exercise in this file. Make the tests in
`space_age_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/space-age` directory.
=end

class SpaceAge
  # add support for age attr
  attr_accessor :age

  # define seconds per year
  SECONDS_PER_YEAR = 31557600

  def initialize(age)
    @age = age
  end

  def on_earth
    # convert seconds to years with floating point division
    @age.fdiv(SECONDS_PER_YEAR)
  end

  def on_mercury
    # convert seconds to years with floating point division, then convert Earth to planet

    @age.fdiv(SECONDS_PER_YEAR).fdiv(0.2408467)
  end

  def on_venus
    # convert seconds to years with floating point division, then convert Earth to planet
    @age.fdiv(SECONDS_PER_YEAR).fdiv(0.61519726)
  end

  def on_mars
    # convert seconds to years with floating point division, then convert Earth to planet
    @age.fdiv(SECONDS_PER_YEAR).fdiv(1.8808158)
  end

  def on_jupiter
    # convert seconds to years with floating point division, then convert Earth to planet
    @age.fdiv(SECONDS_PER_YEAR).fdiv(11.862615)
  end

  def on_saturn
    # convert seconds to years with floating point division, then convert Earth to planet
    @age.fdiv(SECONDS_PER_YEAR).fdiv(29.447498)
  end

  def on_uranus
    # convert seconds to years with floating point division, then convert Earth to planet
    @age.fdiv(SECONDS_PER_YEAR).fdiv(84.016846)
  end

  def on_neptune
    # convert seconds to years with floating point division, then convert Earth to planet
    @age.fdiv(SECONDS_PER_YEAR).fdiv(164.79132)
  end

end

File: solutions/python/space-age/1/space_age.py
class SpaceAge:
    # define seconds per year
    SECONDS_PER_YEAR = 31557600
    # define significant digits for tests
    SIG_DIGS = 2

    # initialize object with seconds
    def __init__(self, seconds):
        # initialize seconds
        self.seconds = seconds

    def on_earth(self):
        return round(self.seconds / self.SECONDS_PER_YEAR, self.SIG_DIGS)

    def on_mercury(self):
        return round(self.seconds / (self.SECONDS_PER_YEAR * 0.2408467), self.SIG_DIGS)

    def on_venus(self):
        return round(self.seconds / (self.SECONDS_PER_YEAR * 0.61519726), self.SIG_DIGS)

    def on_mars(self):
        return round(self.seconds / (self.SECONDS_PER_YEAR * 1.8808158), self.SIG_DIGS)

    def on_jupiter(self):
        return round(self.seconds / (self.SECONDS_PER_YEAR * 11.862615), self.SIG_DIGS)

    def on_saturn(self):
        return round(self.seconds / (self.SECONDS_PER_YEAR * 29.447498), self.SIG_DIGS)

    def on_uranus(self):
        return round(self.seconds / (self.SECONDS_PER_YEAR * 84.016846), self.SIG_DIGS)

    def on_neptune(self):
        return round(self.seconds / (self.SECONDS_PER_YEAR * 164.79132), self.SIG_DIGS)